pkg/networkservice/chains/nsmgr/vl3_test.go

```diff
@@ -509,3 +511,73 @@ func Test_FloatingInterdomain_vl3_dns(t *testing.T) {
 	_, err = resolver.LookupIP(ctx, "ip4", fmt.Sprintf("%s.%s", nscName, searchDomain))
 	require.Error(t, err)
 }
+
+func Test_NSC_ConnectsTo_vl3NSE_With_Invalid_IpContext(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	domain := sandbox.NewBuilder(ctx, t).
+		SetNodesCount(1).
+		SetNSMgrProxySupplier(nil).
+		SetRegistryProxySupplier(nil).
+		Build()
+
+	nsRegistryClient := domain.NewNSRegistryClient(ctx, sandbox.GenerateTestToken)
+
+	nsReg, err := nsRegistryClient.Register(ctx, defaultRegistryService("vl3"))
+	require.NoError(t, err)
+
+	nseReg := defaultRegistryEndpoint(nsReg.Name)
+
+	var serverPrefixCh = make(chan *ipam.PrefixResponse, 1)
+	defer close(serverPrefixCh)
+
+	prefix1 := "10.0.0.0/24"
+	prefix2 := "10.10.0.0/24"
+
+	serverPrefixCh <- &ipam.PrefixResponse{Prefix: prefix1}
+	strictIPPool := ippool.NewWithNetString(prefix1)
+
+	_ = domain.Nodes[0].NewEndpoint(
+		ctx,
+		nseReg,
+		sandbox.GenerateTestToken,
+		strictvl3ipam.NewServer(ctx, vl3.NewServer(ctx, serverPrefixCh), strictIPPool),
```

Comment:
can we make it easier to use?

For example:
`strictvl3ipam.NewServer(ctx, vl3.NewServer, ipPool)`
Like this
https://github.com/networkservicemesh/sdk/blob/b1a3e26da8573763bc4c200d2aaffb873c7947db/pkg/networkservice/chains/nsmgr/suite_test.go#L244